servicehandler/portsscan.go

```diff
@@ -0,0 +1,50 @@
+package servicehandler
+
+import (
+	"context"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/kubescape-network-scanner/cmd"
+	v1 "k8s.io/api/core/v1"
+)
+
+type Port struct {
+	port              int
+	protocol          string
+	sessionLayer      string
+	presentationLayer string
+	applicationLayer  string
+	authenticated     bool
+}
+
+func (port *Port) scan(ctx context.Context, ip string) {
+	result, err := cmd.ScanTargets(ctx, ip, port.port)
+	port.applicationLayer = result.ApplicationLayer
+	port.presentationLayer = result.PresentationLayer
+	port.sessionLayer = result.SessionLayer
+	port.authenticated = result.IsAuthenticated
+
+	if result.ApplicationLayer == "" {
+		// if we can't get the application layer, then we change to Unknown
```

Comment:
debug message would be nice

Reply:
dont agree there is no enough context for the user , 
and in the scanner POV if i cant discover the app its mean it is safe from me
th elogic for this will be in the control